### PR TITLE
Adding additional source and fallback folder properties

### PR DESCRIPTION
**Pull request by emgarten**
This adds support for reading warning properties from packagereferences and the project file on the command line. 

* Adding warning and errors properties
* Adding additional source and fallback folder properties

### CODE REVIEW

**Review by nkolev92 — APPROVED** [review]

File: src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs [hunk 1]
```diff
@@ -715,12 +751,27 @@ private static bool IsPersistDGSet()
...
+        /// Splits and parses a ; or , delimited list of log codes.
+        /// Ignores codes that are unknown.
+        /// </summary>
+        public static IEnumerable<NuGetLogCode> GetNuGetLogCodes(string s)
+        {
+            foreach (var item in MSBuildStringUtility.Split(s, ';', ','))
+            {
+                if (Enum.TryParse<NuGetLogCode>(item, out var result))
+                {
+                    yield return result;
```
Comment (mishra14) [inline src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs:772]: 👍

File: test/NuGet.Core.Tests/NuGet.Commands.Test/MSBuildRestoreUtilityTests.cs [hunk 2]
```diff
@@ -1459,6 +1461,238 @@ public void MSBuildRestoreUtility_ContainsClearKeyword(string input, bool expect
...
+                var wrappedItems = items.Select(CreateItems).ToList();
+
+                // Act
+                var dgSpec = MSBuildRestoreUtility.GetDependencySpec(wrappedItems);
+                var project1Spec = dgSpec.Projects.Single();
+                var props = project1Spec.RestoreMetadata.ProjectWideWarningProperties;
+
+                // Assert
+                props.AllWarningsAsErrors.Should().BeTrue();
+                props.NoWarn.ShouldBeEquivalentTo(new[] { NuGetLogCode.NU1100, NuGetLogCode.NU1101 });
```
Comment (mishra14) [inline test/NuGet.Core.Tests/NuGet.Commands.Test/MSBuildRestoreUtilityTests.cs:1508]: 💯

**Review by mishra14 — APPROVED** [review]
Looks good. We should add E2E or functional tests for this stuff.

**Comment by emgarten** [comment]
Merged

**Pull request closed without merging**